### ISSUE TITLE
Convert basic operators

### ISSUE THREAD

**Issue by angelolloqui** [angelolloqui/SwiftKotlin#43]
Swift allows for custom operators. In Kotlin, operators are restricted to:
https://kotlinlang.org/docs/reference/operator-overloading.html

Transform the operators that have a Kotlin counterpart. Example:

```
public func + (left: Price, right: Price) -> Price {
    return Price(amount: left.amount + right.amount, currency: left.currency)
}
public func - (left: Price, right: Price) -> Price {
    return Price(amount: left.amount - right.amount, currency: left.currency)
}
```

Translates to:
```
operator fun Price.plus(price: Price) =
        Price(amount = this.amount + price.amount, currency = this.currency)

operator fun Price.minus(price: Price) =
        Price(amount = this.amount - price.amount, currency = this.currency)
```

**Comment by angelolloqui** [comment]
Continuing conversation from #89, I believe this conversion can be added by:

1. Changing the swift func into a method extension on the first argument type
2. Remove the first argument and replace tokens with its name by copies with `this` 
3. Add the `operator` keyword
4. Map basic operators to their names (`+` -> `plus`, `-` -> `minus`,... https://kotlinlang.org/docs/reference/operator-overloading.html)

**Comment by torlangballe** [comment]
This would be great. Since they were outside of class/struct in swift I just went for a simple hack, but this would be clean and perfect. Maybe you do it?

**Comment by angelolloqui** [comment]
Yeah, it is in the scope of changes I want to make, but to be honest, it has been open for more than 1 year already. I am not finding all the time I would like to implement all the issues (and the list is long). So yes, I hope I will eventually implement it myself but I can not promise when. 

If this is important for you, please let me know so I will try to prioritize over others

**Comment by torlangballe** [comment]
I can work my way through all other changes first. Can always do a pull request and see if I can do it how you want too if it gets to that, but no hurry yet.

tor
